### PR TITLE
Update prometheus metrics names

### PR DESCRIPTION
**Pull request by hasit**
- Follow prometheus recommended conventions for metrics name
- Update usage throughout aperture

- - -
This change is [Reviewable](https://reviewable.io/reviews/fluxninja/aperture/386)

### CODE REVIEW

File: pkg/metrics/schema.go [hunk 1]
```diff
@@ -3,38 +3,40 @@ package metrics
...
+	// HTTPErrorMetricName - metric from http server.
+	HTTPErrorMetricName = "http_errors_total"
+
+	// Circuit metrics.
+
+	// CircuitSignalReadingMetricName - used in circuit metrics.
+	CircuitSignalReadingMetricName = "circuit_signal_reading"
 	// FluxMeterMetricName name of fluxmeter metrics.
-	FluxMeterMetricName = "flux_meter"
+	FluxMeterMetricName = "flux_meter_info"
```
Comment (tanveergill) [inline pkg/metrics/schema.go:20]: Why flux_meter_info ??

File: pkg/metrics/schema.go [hunk 2]
```diff
@@ -3,38 +3,40 @@ package metrics
...
-	// LatencyHistogramMetricName - metric from http server.
-	LatencyHistogramMetricName = "http_server_request_latency_seconds"
-	// AcceptedConcurrencyMetricName - metric for measuring latencies of accepted requests.
-	AcceptedConcurrencyMetricName = "accepted_concurrency"
-	// IncomingConcurrencyMetricName - metric for measuring latencies of all incoming requests.
-	IncomingConcurrencyMetricName = "incoming_concurrency"
+	WorkloadLatencyMetricName = "workload_latency_seconds"
 	// WorkloadLatencySumMetricName - metric from workload histogram.
-	WorkloadLatencySumMetricName = "workload_latency_ms_sum"
+	WorkloadLatencySumMetricName = "workload_latency_seconds_sum"
```
Comment (tanveergill) [inline pkg/metrics/schema.go:27]: this should be ms not seconds!!

File: pkg/metrics/schema.go (hunk 1 above)
Comment (hasit) [reply pkg/metrics/schema.go]: Changed it to `flux_meter`

File: pkg/metrics/schema.go (hunk 2 above)
Comment (hasit) [reply pkg/metrics/schema.go]: Done.

File: pkg/metrics/schema.go [hunk 3]
```diff
@@ -3,38 +3,40 @@ package metrics
...
 	// METRIC NAMES.
 
-	// SignalReadingMetricName - used in circuit metrics.
-	SignalReadingMetricName = "signal_reading"
+	// HTTP metrics.
+
+	// HTTPRequestMetricName - metric from http server.
+	HTTPRequestMetricName = "http_requests_total"
+	// HTTPRequestLatencyMetricName - metric from http server.
+	HTTPRequestLatencyMetricName = "http_requests_latency_seconds"
```
Comment (tanveergill) [inline pkg/metrics/schema.go:11]: This is ms as well. Always refer [where its used](https://github.com/fluxninja/aperture/blob/828775e24d588c26e7a967bbbe9adb4676c8a203/pkg/net/http/server.go#L224) to double check

File: pkg/metrics/schema.go [hunk 4]
```diff
@@ -3,38 +3,40 @@ package metrics
...
+	// HTTPErrorMetricName - metric from http server.
+	HTTPErrorMetricName = "http_errors_total"
+
+	// Circuit metrics.
+
+	// CircuitSignalReadingMetricName - used in circuit metrics.
+	CircuitSignalReadingMetricName = "circuit_signal_reading"
```
Comment (tanveergill) [inline pkg/metrics/schema.go:18]: Harjot and I like `signal_reading` better, please revert

File: pkg/metrics/schema.go [hunk 5]
```diff
@@ -3,38 +3,40 @@ package metrics
...
-	AcceptedConcurrencyMetricName = "accepted_concurrency"
-	// IncomingConcurrencyMetricName - metric for measuring latencies of all incoming requests.
-	IncomingConcurrencyMetricName = "incoming_concurrency"
 	// WorkloadLatencySumMetricName - metric from workload histogram.
 	WorkloadLatencySumMetricName = "workload_latency_ms_sum"
 	// WorkloadLatencyCountMetricName - metric from workload histogram.
 	WorkloadLatencyCountMetricName = "workload_latency_ms_count"
+
+	// AcceptedConcurrencyMetricName - metric for measuring latencies of accepted requests.
+	AcceptedConcurrencyMetricName = "accepted_concurrency_total"
```
Comment (tanveergill) [inline pkg/metrics/schema.go:32]: Concurrency is `ms`!!

File: pkg/metrics/schema.go [hunk 6]
```diff
@@ -3,38 +3,40 @@ package metrics
...
-	FlowControlRequestsMetricName = "flowcontrol_requests_count"
-	// FlowControlDecisionsMetricName - counter for Check requests per decision type.
-	FlowControlDecisionsMetricName = "flowcontrol_decisions_count"
-	// FlowControlErrorReasonMetricName - metric for error reason on FCS Check requests.
-	FlowControlErrorReasonMetricName = "flowcontrol_error_reason_count"
-	// FlowControlRejectReasonMetricName - metric for reject reason on FCS Check requests.
-	FlowControlRejectReasonMetricName = "flowcontrol_reject_reason_count"
+	WFQRequestsMetricName = "wfq_requests_total"
 	// TokenBucketMetricName - a gauge that tracks the load shed factor.
 	TokenBucketMetricName = "token_bucket_lsf"
```
Comment (tanveergill) [inline pkg/metrics/schema.go:41]: Load shed factor is a `ratio`. Refer [prom unit conventions](https://prometheus.io/docs/practices/naming/)

**Comment by hasit** [comment]
@tanveergill suggestion for naming fill rate? I am unclear as it is tokens/sec.

**Comment by tanveergill** [comment]
> @tanveergill suggestion for naming fill rate? I am unclear as it is tokens/sec.

Its technically `_ms_per_second` but I would leave it at rate and mention in the help string that it is tokens per second where the tokens are measured in ms

**Comment by tanveergill** [comment]
@hasit,
Do update the blueprints and any prom dashboards in cloud if any

**Comment by hasit** [comment]
@tanveergill yes, I have pushing them now.

**Review by tanveergill — APPROVED** [review]

**Comment by tanveergill** [comment]
@hasit,

Please make sure we follow OTEL semantics for standard labels like status code on flux meter metric: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attributes

**Comment by hasit** [comment]
@tanveergill at first glance they look fine - https://github.com/fluxninja/aperture/blob/main/pkg/otelcollector/consts.go

**Comment by tanveergill** [comment]
@hasit,

FluxMeter is tagged by `status_code` instead of `http.status_code`. Please double check using gopls/tags to backtrack all the label definitions from all the prom.Register calls in the code

**Review by tanveergill — APPROVED** [review]

**Comment by tanveergill** [comment]
Decided to replace `.`'s with `_` in OTEL recommended metric labels to be compatible with Prometheus

**Comment by tanveergill** [comment]
@hasit,
Please merge

**Pull request merged**